File: .scripts/Gems/GemInfo.ts
export type AllomancyGroup = 'physical' | 'mental' | 'enhancement' | 'temporal';
export type FeruchemyGroup = 'physical' | 'cognitive' | 'spiritual' | 'hybrid';
export type Axis = 'internal' | 'external';
export type Polarity = 'pushing' | 'pulling';

export class GemInfo {
    public name: string;
    public descriptions: { raw: string; cut: string; sphere: string; mining?: string; };
    public color: [number, number, number];
    public colorTwo?: [number, number, number];
    public glowColor: [number, number, number];
    public stackable: boolean = false;
    public drawSize: number = 1;
    public baseBeauty: number;
    public baseMarketValue: number;
    public baseStormlight: number;
    public mining?: MiningInfo;

    constructor(self: Partial<GemInfo>) {
        Object.assign(this, self);
    }
}

export class MiningInfo {
    public hitPoints: number;
    public yield: number;
    public commonality: number;
    public sizeRange: [number, number];

    constructor(self: Partial<MiningInfo>) {
        Object.assign(this, self);
    }
}

File: tools/Cosmere.Tools/Models/GemInfo.cs
namespace Cosmere.Tools.Models;

public class GemInfo
{
    public string Name { get; set; } = string.Empty;
    public GemDescriptions Descriptions { get; set; } = new();
    public string? DefName { get; set; }
    public ColorInfo Color { get; set; } = new();
    public ColorInfo? ColorTwo { get; set; }
    public ColorInfo? GlowColor { get; set; }
    public bool Stackable { get; set; } = true;
    public float DrawSize { get; set; } = 1f;
    public float BaseBeauty { get; set; }
    public float BaseMarketValue { get; set; }
    public int BaseStormlight { get; set; }
    public int MaxAmount { get; set; } = 100;
    public Dictionary<string, string[]> GenesToGrant { get; set; } = new();
    public BuildableInfo? Buildable { get; set; }
    public MiningInfo? Mining { get; set; }
}